src/modules/endpoint/sagas/fetchCategory.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import type { Saga } from 'redux-saga'
+import { call, put, takeLatest, fork, select } from 'redux-saga/effects'
+import type {
+  PushCategoryActionType,
+  FetchCategoryActionType,
+  FetchCategoryFailedActionType,
+  SwitchCategoryLanguageActionType
+} from '../../app/StoreActionType'
+import MemoryDatabase from '../MemoryDatabase'
+import {
+  categoriesEndpoint,
+  CategoriesMapModel,
+  LanguageModel,
+  languagesEndpoint
+} from '@integreat-app/integreat-api-client'
+import request from '../request'
+import type { ResourceCacheType } from '../ResourceCacheType'
+import findResources from '../findResources'
+import fnv from 'fnv-plus'
+import { OFFLINE_CACHE_PATH } from '../../platform/constants/webview.ios'
+import getExtension from '../getExtension'
+import DataContext from '../DataContext'
+import fetchResourceCache from './fetchResourceCache'
+import persistCategories from './persistCategories'
+import type { StateType } from '../../app/StateType'
+
+function * fetchCategoriesByLanguage (city: string, code: string): Saga<CategoriesMapModel> {
+  const params = {
+    city,
+    language: code
+  }
+
+  const categoriesPayload = yield call(() => request(categoriesEndpoint, params))
+  return categoriesPayload.data
+}
+
+function * fetchLanguages (city: string): Saga<Array<LanguageModel>> {
+  const params = {city}
+
+  const languagesPayload = yield call(() => request(languagesEndpoint, params))
```

Comment:
do we need this call(...) because request returns a promise and is not a generator function? (I don't know how saga works^^)

Reply:
Yes, if you want to resolve a  promise or saga you have to use call.